wikimedia/wikipedia-ios
# Set up Notification Service Extension

**Phabricator:** https://phabricator.wikimedia.org/T287310

### Notes
This PR sets up the service extension to populate push content. It adds the extension to the project, updates authentication cookies to use a shared storage in the app container, and makes the fetch call created in #4020 to confirm an authenticated call works from the service extension.

After review, please wait for #4020 to be reviewed and merged. Then update this with feature/notifications and merge into that feature branch.

### Test Steps
1. Launch and log in from `main`, then launch again and run from this PR branch. Confirm authenticated calls still work from the app (you can confirm by editing a test article).
2. Go to Settings > Notifications and enable notifications. After allowing permissions, watch the traffic via a proxy for the subscription call. Use the `providertoken` and `topic` values from this call in the push notification utility.
3. Use the push notification utility to push a notification to your device. Watch the traffic as the push appears via a proxy and confirm a call to the notifications API is made, and properly goes through without a "you must log in" error in the response. Also confirm the push notification displays something like "checkEchoV1 [modified]" to confirm the service extension is modifying the content.
